### PR TITLE
[Flow EVM] add uuids to the deposit and withdraw events

### LINKED ISSUE #6103: [Flow EVM] update token deposit and withdrawn events to include the uuid of vaults
To make tracking of Flow tokens transfers across the environments, ideally we should include the uuid of the vault in the events that we are sending. 

Thread: https://discord.com/channels/613813861610684416/1251170631928778752

### PR DESCRIPTION
**Pull request by ramtinms**
closes #6103

### CODE REVIEW

File: fvm/evm/stdlib/contract.cdc [hunk 1]
```diff
@@ -77,14 +77,26 @@ contract EVM {
...
     /// for transfer of flow tokens between two EVM addresses.
+    /// Similar to the FungibleToken.Deposited event
+    /// this event includes a depositedUUID that captures the 
+    /// uuid of the source vault.
     access(all)
-    event FLOWTokensDeposited(address: String, amount: UFix64)
+    event FLOWTokensDeposited(
+        address: String, 
+        amount: UFix64, 
+        depositedUUID: UInt64)
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:87]:
```suggestion
        depositedUUID: UInt64
   )
```

File: fvm/evm/stdlib/contract.cdc [hunk 2]
```diff
@@ -77,14 +77,26 @@ contract EVM {
...
     /// for transfer of flow tokens between two EVM addresses.
+    /// similar to the FungibleToken.Withdrawn events
+    /// this event includes a withdrawnUUID that captures the 
+    /// uuid of the returning vault.
     access(all)
-    event FLOWTokensWithdrawn(address: String, amount: UFix64)
+    event FLOWTokensWithdrawn(
+        address: String, 
+        amount: UFix64, 
+        withdrawnUUID: UInt64)
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:99]:
```suggestion
        withdrawnUUID: UInt64
    )
```

File: fvm/evm/stdlib/contract.cdc [hunk 3]
```diff
@@ -152,11 +164,12 @@ contract EVM {
             if amount == 0.0 {
                 panic("calling deposit function with an empty vault is not allowed")
             }
+            let depositedUUID = from.uuid
             InternalEVM.deposit(
                 from: <-from,
                 to: self.bytes
             )
-            emit FLOWTokensDeposited(address: self.toString(), amount: amount)
+            emit FLOWTokensDeposited(address: self.toString(), amount: amount, depositedUUID: depositedUUID)
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:172]:
```suggestion
            emit FLOWTokensDeposited(
                address: self.toString(), 
                amount: amount, 
                depositedUUID: depositedUUID
            )
```

File: fvm/evm/stdlib/contract.cdc [hunk 4]
```diff
@@ -383,7 +396,7 @@ contract EVM {
                 from: self.addressBytes,
                 amount: balance.attoflow
             ) as! @FlowToken.Vault
-            emit FLOWTokensWithdrawn(address: self.address().toString(), amount: balance.inFLOW())
+            emit FLOWTokensWithdrawn(address: self.address().toString(), amount: balance.inFLOW(), withdrawnUUID: vault.uuid)
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:399]:
```suggestion
            emit FLOWTokensWithdrawn(
                address: self.address().toString(),
                amount: balance.inFLOW(),
                withdrawnUUID: vault.uuid
            )
```

**Review by turbolent — APPROVED** [review]
👍 

Just a few minor formatting suggestions

**Comment by turbolent** [comment]
Needs an update of the state commitments

File: fvm/evm/stdlib/contract.cdc (hunk 1 above)
Comment (ramtinms) [reply fvm/evm/stdlib/contract.cdc]: updated

**Comment by ramtinms** [comment]
Correct

**Review by sisyphusSmiling — APPROVED** [review]
👍 New event values LGTM

**Pull request merged**